### ISSUE TITLE
[Request] Manual restore without loading new windows

### ISSUE THREAD

**Issue by ziplock9000** [kangyu-california/PersistentWindows#73]
Can we have an option to manually restore windows without opening new ones? So just restore windows that are aleady open?

Hopefully not by some key combination, but with its own menu option?

**Comment by kangyu-california** [comment]
I think snapshot feature would satisfy this requirement, 
snapshot function is invoked by using combination of shift and ctrl key while clicking PW icon,
the usage is documented in readme as well as online help (https://github.com/kangyu-california/PersistentWindows/blob/master/README.md)

**Comment by ziplock9000** [comment]
Can we have this as a menu option instead of having to remember two keys to hold down when clicking? It's a bit 1980 using key combos like that.

**Comment by kangyu-california** [comment]
Clicking icon is faster than invoking multi-level menus.., 

Anyway I will add menu option to capture/restore snapshot

**Comment by jimboca** [comment]
> Anyway I will add menu option to capture/restore snapshot

Thank you!  I can't always remember which combos work for what since I don't need it that often.

**Comment by DannySwish** [comment]
Is there a way to just have each individual program's window positions be remembered and updated in a log without the need for user interaction? I find that windows sometimes don't restore, and if I try restoring manually with the windows saved to disk, it opens other programs that I don't want open at that time. I understand your solution to this was the snapshots feature, but again, this involves making presets and remembering them, and then manually selecting them. It's too much work and cannot feasibly cover every single combination of open/closed programs.

Here's an example of what I just experienced:

I have a saved capture to disk with all of my programs that I like open opened. This works fine when these are the windows I want open and in the correct positions.

However, I wanted to close some of these programs because I didn't want them at the time. The problem that arose is that, after display sleep and wake-up rearranging everything, the programs that I had left open didn't all successfully move to their correct positions automatically. I guess PersistentWindows failed that time. The problem is, selecting to restore from disk reopens (and in some cases opens new instances) of the programs I wanted to stay closed. But this was the only place I had the configurations for the windows saved, as I did not think to pre-make a snapshot for what I was working with at the time. From what I understand, I would have to make a snapshot for every single combination of windows I want to preserve, instead of programs just "remembering" where they were the last time they were in use.

What I suggest is somehow pulling the latest user-made modifications to a window's size and position for each window of a program, and storing that in a repository to pull from the next time that program is either launched or found to be in the incorrect position/dimensions, instead of manually grouping all current window sizes and placements into snapshots. Ideally, nobody would ever have to directly interact with the PersistentWindows program itself, as it would manage everything behind the scenes (perhaps with the occasional need for a button which examines which programs/windows are open and reapplies the last known user-made placements and dimensions to those windows, eliminating the need for snapshots and captures to disk).

I'm not a programmer, so I apologize in advance if what I'm suggesting isn't practical or achievable. But this would once and for all resolve the issue.
